File: CodeForces/362/Main2.java
import java.util.*;
import java.util.regex.*;
import java.math.*;
import java.io.*;

public class Main2 {
	public static void main(String args[]) {
		Scanner sc = new Scanner(System.in);

		Pattern p = Pattern.compile("((\\d).(\\d+))e(\\d+)");
		String inp = sc.nextLine();

		Matcher m = p.matcher(inp);
		boolean matches = m.matches();

		if(m.group(3).equals("0") && m.group(4).equals("0"))
		{
			System.out.println(m.group(2));
		}
		else if(matches)
		{
			BigDecimal num = new BigDecimal(m.group(1));
			BigDecimal exp = new BigDecimal(m.group(4));

			BigDecimal ans = num.scaleByPowerOfTen(exp.intValue());

			System.out.println(ans.toPlainString());
		}
	}
}

File: CodeForces/362/main2.cpp
#include <iostream>
#include <regex>

using namespace std;

int main() 
{
	int exp, frac1, i=0;
	char frac2[10000] = {'0'};

	scanf("%d.", &frac1);

	scanf("%c", &frac2[i]);
	while(frac2[i] != 'e')
	{
		i++;
		scanf("%c", &frac2[i]);
	}

	scanf("%d", &exp);

	i=0;
	cout << frac1;
	
	while(true)
	{
		if(i<exp)
		{
			cout << frac2[i];
		}
		else if(frac2[i] == 'e')
		{
			cout << ".";
			while(i<exp)
			{
				cout << 0;
				i++;
			}
			break;
		}
		
		

		i++;
	}

	cout << endl;
}